server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/ToRecipientFolderTest.java

```diff
@@ -77,8 +77,7 @@ public void setUp() throws Exception {
         mailboxManager = mock(MailboxManager.class);
 
 
-        MetricFactory metricFactory = mock(MetricFactory.class);
-        when(metricFactory.generate(anyString())).thenReturn(mock(Metric.class));
+        metricFactory = new RecordingMetricFactory();
```

Comment:
Same thing here, these tests relies too much on mocks to be consensual. Metrics were not tested. Using `NoopMetricFactory` here would be a good move.